### ISSUE TITLE
Valid sstables are identified as invalid by scrub compaction in validate mode

### ISSUE THREAD

**Issue by bhalevy** [scylladb/scylladb#13832]
This regression was caused by 5f37d43ee627b70807bd08ca810281677ea996e8
Found in https://jenkins.scylladb.com/view/master/job/scylla-master/job/dtest-daily-release/245/testReport/compaction_additional_test/TestValidationCompaction/Run_Dtest_Parallel_Cloud_Machines___FullDtest___full_split000___test_validation_compaction_with_valid_sstable/

See https://jenkins.scylladb.com/view/master/job/scylla-master/job/dtest-daily-release/245/artifact/logs-full.release.000/1683684292866_compaction_additional_test.py%3A%3ATestValidationCompaction%3A%3Atest_validation_compaction_with_valid_sstable/node1.log
```
INFO  2023-05-10 02:04:49,751 [shard 1] compaction - Scrubbing in validate mode /jenkins/workspace/scylla-master/dtest-daily-release/scylla/.dtest/dtest-tw42y47f/test/node1/data/ks/cf-074cb3c0eed711ed908e9cf6445bdd95/me-3-big-Data.db
INFO  2023-05-10 02:04:49,751 [shard 0] compaction - Scrubbing in validate mode /jenkins/workspace/scylla-master/dtest-daily-release/scylla/.dtest/dtest-tw42y47f/test/node1/data/ks/cf-074cb3c0eed711ed908e9cf6445bdd95/me-2-big-Data.db
INFO  2023-05-10 02:04:49,757 [shard 1] compaction - Finished scrubbing in validate mode /jenkins/workspace/scylla-master/dtest-daily-release/scylla/.dtest/dtest-tw42y47f/test/node1/data/ks/cf-074cb3c0eed711ed908e9cf6445bdd95/me-3-big-Data.db - sstable is invalid
INFO  2023-05-10 02:04:49,757 [shard 0] compaction - Finished scrubbing in validate mode /jenkins/workspace/scylla-master/dtest-daily-release/scylla/.dtest/dtest-tw42y47f/test/node1/data/ks/cf-074cb3c0eed711ed908e9cf6445bdd95/me-2-big-Data.db - sstable is invalid
INFO  2023-05-10 02:04:49,757 [shard 1] sstable - Moving sstable /jenkins/workspace/scylla-master/dtest-daily-release/scylla/.dtest/dtest-tw42y47f/test/node1/data/ks/cf-074cb3c0eed711ed908e9cf6445bdd95/me-3-big-Data.db to "/jenkins/workspace/scylla-master/dtest-daily-release/scylla/.dtest/dtest-tw42y47f/test/node1/data/ks/cf-074cb3c0eed711ed908e9cf6445bdd95/quarantine"
INFO  2023-05-10 02:04:49,757 [shard 0] sstable - Moving sstable /jenkins/workspace/scylla-master/dtest-daily-release/scylla/.dtest/dtest-tw42y47f/test/node1/data/ks/cf-074cb3c0eed711ed908e9cf6445bdd95/me-2-big-Data.db to "/jenkins/workspace/scylla-master/dtest-daily-release/scylla/.dtest/dtest-tw42y47f/test/node1/data/ks/cf-074cb3c0eed711ed908e9cf6445bdd95/quarantine"
```

For some reason scrub compaction in validate mode decides that the "sstable is invalid", without printing any specific validation error.

The test pass at b790f144567dd5c4c4d15b71cd5cdd0cf5822ba9 and fails this way at 5f37d43ee627b70807bd08ca810281677ea996e8

**Comment by denesb** [comment]
I think this is https://github.com/scylladb/scylladb/pull/13813. I will check if https://github.com/scylladb/scylladb/commit/5f37d43ee627b70807bd08ca810281677ea996e8 is indeed before the fix.

**Comment by denesb** [comment]
> I think this is #13813. I will check if [5f37d43](https://github.com/scylladb/scylladb/commit/5f37d43ee627b70807bd08ca810281677ea996e8) is indeed before the fix.

Oh, https://github.com/scylladb/scylladb/commit/5f37d43ee627b70807bd08ca810281677ea996e8 is right at where the PR was merged. So the fix was not in yet (was merged yesterday).

**Comment by bhalevy** [comment]
Yup, fixed by d8cd62b91ae446d5c0caee381c136e313a50b043
